hibernate-core/src/test/java/org/hibernate/orm/test/serialization/entity/PK.java

```diff
@@ -30,4 +32,24 @@ protected Long getValue() {
 	protected void setValue(Long value) {
 		this.value = value;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if ( this == o ) {
+			return true;
+		}
+		if ( o == null || getClass() != o.getClass() ) {
+			return false;
+		}
+
+		PK pk = (PK) o;
+
+		return !( value != null ? !value.equals( pk.value ) : pk.value != null );
```

Comment:
we might well use the new util method `Objects#equals(value, pk.value)` from JDK8 and `Objects#hashCode(value)` below to save some chore and ensure nullness correctness, but it would shine more for multiple-field case.